libs/s25main/ingameWindows/iwStatistics.cpp

```diff
@@ -278,29 +279,62 @@ void iwStatistics::Draw_()
     DrawStatistic(currentView);
 }
 
-void iwStatistics::DrawPlayerBoxes()
+void iwStatistics::DrawPlayerOverlays()
 {
-    unsigned short startX = 126 - numPlayingPlayers * 17;
-    DrawPoint drawPt = GetDrawPos() + DrawPoint(startX, 68);
+    unsigned short startX = 126 - (numPlayingPlayers - 1) * 17;
+    DrawPoint drawPt = GetDrawPos() + DrawPoint(startX - 17, 45 - 23);
     for(unsigned i = 0; i < gwv.GetWorld().GetNumPlayers(); ++i)
     {
         const GamePlayer& player = gwv.GetWorld().GetPlayer(i);
         if(!player.isUsed())
             continue;
 
-        if(activePlayers[i])
-        {
-            auto const playerBoxRect = Rect(drawPt, Extent(34, 12));
-            auto const playerStatusPosition =
-              DrawPoint(playerBoxRect.getOrigin() + playerBoxRect.getSize() / 2 + DrawPoint(0, 1));
-            DrawRectangle(playerBoxRect, player.color);
-            SmallFont->Draw(playerStatusPosition, getPlayerStatus(player), FontStyle::CENTER | FontStyle::VCENTER,
-                            COLOR_YELLOW);
-        }
+        // Draw the alliances at top of the player image
+        DrawPlayerAlliances(drawPt, i);
+        // Draw player boxes and player status at bottom
+        DrawPlayerBoxes(drawPt, player);
         drawPt.x += 34;
     }
 }
 
+void iwStatistics::DrawPlayerBoxes(DrawPoint const& drawPt, const GamePlayer& player)
+{
+    auto playerId = player.GetPlayerId();
+    if(activePlayers[playerId])
+    {
+        auto const playerBoxRect = Rect(DrawPoint(drawPt.x, drawPt.y + 47), Extent(34, 12));
+        auto const playerStatusPosition =
+          DrawPoint(playerBoxRect.getOrigin() + playerBoxRect.getSize() / 2 + DrawPoint(0, 1));
+        DrawRectangle(playerBoxRect, player.color);
+        SmallFont->Draw(playerStatusPosition, getPlayerStatus(player), FontStyle::CENTER | FontStyle::VCENTER,
+                        COLOR_YELLOW);
+    }
+}
+
+void iwStatistics::DrawPlayerAlliances(DrawPoint const& drawPt, unsigned playerId)
+{
+    constexpr unsigned spacingAllianceRects = 1;
+    constexpr Extent allianceRectExtent(4, 6);
+    constexpr unsigned allianceRectOffset = allianceRectExtent.x + spacingAllianceRects;
+
+    const GameWorldBase& world = gwv.GetWorld();
+    const GamePlayer& curPlayer = world.GetPlayer(playerId);
+
+    auto pactPos = drawPt + DrawPoint(spacingAllianceRects, spacingAllianceRects);
+    for(auto idxOther : helpers::range(world.GetNumPlayers()))
+    {
+        if(idxOther == playerId)
+            continue;
+
+        const GamePlayer& otherPlayer = world.GetPlayer(idxOther);
+        if(!curPlayer.IsAlly(idxOther))
+            continue;
+
+        DrawRectangle(Rect(pactPos, allianceRectExtent), otherPlayer.color);
+        pactPos.x += allianceRectOffset;
```

Comment:
Even easier now and more readable:
```suggestion
        if(idxOther != playerId && curPlayer.IsAlly(idxOther))
        {
            DrawRectangle(Rect(pactPos, allianceRectExtent), world.GetPlayer(idxOther).color);
            pactPos.x += allianceRectOffset;
        }
```